erts/emulator/test/match_spec_SUITE.erl

```diff
@@ -1145,7 +1147,32 @@ moving_labels(Config) when is_list(Config) ->
 	erlang:match_spec_test({true,false}, Ms2, table),
 
     ok.
-    
+
+%% GH-7045: Some guard BIFs were unavailable in match specifications.
+guard_bifs(_Config) ->
+    Matches =
+        [begin
+             BIF = case A of
+                       0 -> {F};
+                       1 -> {F, '$1'};
+                       2 -> {F, '$1', '$2'};
+                       3 -> {F, '$1', '$2', '$3'}
+                   end,
+             erlang:match_spec_test({a, b, c},
+                                    [{{'$1','$2', '$3'},
+                                      [BIF],
+                                      [{{'$1', '$2', '$3'}}]}],
+                                    table)
+         end
+         || {F, A} <- erlang:module_info(functions),
+            F =/= is_record,
```

Comment:
I'd point out that `is_record/3` ***does*** work in match specs, it's only `is_record/2` that does not, so we may want to change this filter a little to reflect that.